WordPress/src/main/java/org/wordpress/android/ui/posts/photochooser/ThumbnailLoader.java

```diff
@@ -0,0 +1,121 @@
+package org.wordpress.android.ui.posts.photochooser;
+
+import android.animation.ObjectAnimator;
+import android.content.Context;
+import android.graphics.Bitmap;
+import android.os.Handler;
+import android.os.Looper;
+import android.provider.MediaStore;
+import android.view.View;
+import android.widget.ImageView;
+
+import org.wordpress.android.R;
+
+import java.lang.ref.WeakReference;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+
+class ThumbnailLoader {
+
+    private final Context mContext;
+    private final ThreadPoolExecutor mExecutor;
+    private final Handler mHandler;
+    private boolean mIsFadeEnabled = true;
+
+    private static final int FADE_TRANSITION = 250;
+
+    ThumbnailLoader(Context context) {
+        mContext = context;
+        int numCores = Runtime.getRuntime().availableProcessors();
+        int maxThreads = numCores > 1 ? numCores / 2 : 1;
+        mExecutor = (ThreadPoolExecutor) Executors.newFixedThreadPool(maxThreads);
+        mExecutor.setRejectedExecutionHandler(new ThreadPoolExecutor.DiscardOldestPolicy());
+        mHandler = new Handler(Looper.getMainLooper());
+    }
+
+    void loadThumbnail(ImageView imageView, long imageId, boolean isVideo) {
+        Runnable task = new ThumbnailLoaderRunnable(imageView, imageId, isVideo);
+        mExecutor.submit(task);
+    }
+
+    /*
+     * temporarily disables the fade animation - called when multiple items are changed
+     * to prevent unnecessary fade/flicker
+     */
+    void temporarilyDisableFade() {
```

Comment:
If I was looking into code only, this looks like an unnecessary hack, but... I love how you take care of all those subtleties! ❤️